src/main/java/com/mongodb/kafka/connect/util/TimeseriesValidation.java

```diff
@@ -209,21 +199,22 @@ private static CreateCollectionOptions createCollectionOptions(
       timeSeriesOptions.metaField(metaField);
     }
 
-    MongoSinkTopicConfig.TimeSeriesGranularity granularity =
-        MongoSinkTopicConfig.TimeSeriesGranularity.valueOf(
-            config.getString(TIMESERIES_GRANULARITY_CONFIG).trim().toUpperCase());
-    switch (granularity) {
-      case SECONDS:
-        timeSeriesOptions.granularity(TimeSeriesGranularity.SECONDS);
-        break;
-      case MINUTES:
-        timeSeriesOptions.granularity(TimeSeriesGranularity.MINUTES);
-        break;
-      case HOURS:
-        timeSeriesOptions.granularity(TimeSeriesGranularity.HOURS);
-        break;
-      default:
-        // Do nothing
+    String granularityValue =
+        config.getString(TIMESERIES_GRANULARITY_CONFIG).trim().toUpperCase(Locale.ROOT);
+    if (!granularityValue.isEmpty()) {
+      switch (TimeSeriesGranularity.valueOf(granularityValue)) {
```

Comment:
This method throws an `IllegalArgumentException` if `granularityValue` cannot be converted to a `TimeSeriesGranularity`. Should it be wrapped into a `ConfigException`?

Reply:
The validators in the MongoSinkTopicConfig ensure that its either an empty string or can be converted.

Reply:
Done

Reply:
> The validators in the MongoSinkTopicConfig ensure that its either an empty string or can be converted.

Sorry, I forgot there is `Validators.emptyString().or(Validators.EnumValidatorAndRecommender.in(TimeSeriesGranularity.values()))` in `MongoSinkTopicConfig`. If I realized that, I would have not left the comment.